source/JABHandler.py

```diff
@@ -519,11 +539,12 @@ def getAccessibleKeyBindings(self):
 
 @AccessBridge_FocusGainedFP
 def internal_event_focusGained(vmID, event,source):
-	internalQueueFunction(event_gainFocus,vmID,source)
+	hwnd=getWindowHandleFromAccContext(vmID,source)
+	internalQueueFunction(event_gainFocus,vmID,source,hwnd)
 	bridgeDll.releaseJavaObject(vmID,event)
 
-def event_gainFocus(vmID,accContext):
-	jabContext=JABContext(vmID=vmID,accContext=accContext)
+def event_gainFocus(vmID,accContext,hwnd):

```

Comment:
I'm probably missing something obvious here, but I don't understand why you've added hwnd arguments to the event_* functions. From what I can see, you always call getWindowHandleFromAccContext to get the hwnd before calling the event, but JABContext will do this anyway if there's no hwnd provided. Can you please explain?